script/DCA_v2.s.sol

```diff
@@ -0,0 +1,220 @@
+        address transformerRegistry = deployCreate3({_name: "TransformerRegistry", _salt: 'MF-Transformer-Registry-V1', _args: abi.encode(msig)});
+        if (block.chainid == 1) {
+            address stETH = 0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84;
+            deployCreate3({_name: "wstETHTransformer", _salt: 'MF-wstETH-Transformer-V1', _args: abi.encode(stETH, msig)});
+        }
+
+        // Oracles
+        address transformerOracle = deployOracles({
+            _transformerRegistry: transformerRegistry,
+            _chainlinkFeedRegistry: chainlinkFeedRegistry
+        });
+
+        // DCA V2 Core
+        address permissionsManager = deployCreate3({_name: 'DCAPermissionsManager', _salt: 'MF-DCAV2-PermissionsManager-V1', _args:  abi.encode(msig, tokenDescriptor)});
+        address timelock = deployCreate3({_name: "TimelockController", _salt: 'MF-DCAV2-Timelock-V1', _args: abi.encode(3 days, msigArray, msigArray)});
+        address dcaHub = deployCreate3({_name: "DCAHub", _salt: 'MF-DCAV2-DCAHub-V1', _args: abi.encode(msig, timelock, transformerOracle, permissionsManager)});
+
+        // Swappers
+        address swapperRegistry = deployCreate3({_name: "SwapperRegistry", _salt: 'MF-Swapper-Registry-V1', _args: abi.encode(new address[](0), new address[](0), msig, msigArray)});
+        deployCreate3({_name: 'SwapProxy', _salt: 'MF-Swap-Proxy-V1', _args: abi.encode(swapperRegistry, msig)});
+
+        // DCA V2 Periphery
+        deployCreate3({_name: 'DCAHubCompanion', _salt: 'MF-DCAV2-DCAHubCompanion-V5', _args: abi.encode(permit2AdapterAddress, permit2AdapterAddress, msig, PERMIT2)});
+        deployCreate3({_name: 'CallerOnlyDCAHubSwapper', _salt: 'MF-DCAV2-CallerDCAHubSwapper-V2', _args: hex""});
+        deployCreate3({_name: 'ThirdPartyDCAHubSwapper', _salt: 'MF-DCAV2-3PartySwapper-V3', _args: hex""});
+        deployCreate3({_name: 'DCAFeeManager', _salt: 'MF-DCAV2-DCAFeeManager-V3', _args: abi.encode(msig, msigArray)});
+        if (block.chainid == 1) {
+            address keep3r = 0xeb02addCfD8B773A5FFA6B9d1FE99c566f8c44CC;
+            deployCreate3({_name: 'DCAKeep3rJob', _salt: 'MF-DCAV2-Keep3rJob-V2', _args: abi.encode(keep3r, dcaHub, msig, new address[](0))});
+        }
+    }
+
+    /// @dev Deploys the Oracle contracts. We use a separate function to prevent a "Stack too deep"
+    /// Solidity compiler error, which would occur if we put these deployments in the main `run`
+    /// function.
+    function deployOracles(address _transformerRegistry, address _chainlinkFeedRegistry) internal returns (address transformerOracle) {
+        address chainlinkOracle = deployCreate3({_name: "StatefulChainlinkOracle", _salt: 'MF-StatefulChainlink-Oracle-V2', _args: abi.encode(_chainlinkFeedRegistry, msig, msigArray)});
+        address uniswapV3Oracle = 0xB210CE856631EeEB767eFa666EC7C1C57738d438;
+        address uniswapV3Adapter;
+        if (uniswapV3Oracle.code.length > 0) {
```

Comment:
This doesn't adhere strictly to the [corresponding hardhat-deploy script](https://github.com/Mean-Finance/oracles/blob/main/deploy/002_uniswap_v3_adapter.ts#L8-L28), where specific networks are hard-coded. I figured this is sufficient for now though. Fyi, the `UniswapV3Oracle` isn't deployed on networks like Optimism Sepolia, which will cause the `UniswapV3Adapter` deployment to fail.